client/src/components/Connections/components/Search.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+import React from 'react'
+import {withRouter} from 'react-router-dom'
+import {withHandlers, withState} from 'recompose'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {updateValue} from '../../../actions/sharedActions'
+import {Form, FormGroup, Label, Input, Button} from 'reactstrap'
+import type {ContextRouter} from 'react-router'
+
+import EntitySearchWrapper, {type EntitySearchProps} from '../dataWrappers/EntitySearchWrapper'
+import './Search.css'
+
+type EmptyHandler = () => void // TODO extract
+type EventHandler<T> = (e: T) => void
+
+const checkEnter = (callback: EmptyHandler): EventHandler<KeyboardEvent> => {
+  return (e: KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      callback()
+    }
+  }
+}
+
+type Props = {
+  searchValue1: string,
+  searchValue2: string,
+  setSearchValue1: (entitySearch: string) => void,
+  setSearchValue2: (entitySearch: string) => void,
+  searchConnection: EmptyHandler,
+} & EntitySearchProps &
+  ContextRouter
+
+const Search = ({
+  searchValue1,
+  searchValue2,
+  setSearchValue1,
+  setSearchValue2,
+  searchConnection,
+}: Props) => (
+  <div>
+    <h2>Vyhľadaj</h2>
+    <p>najkratšie spojenie medzi dvojicou:</p>
+    <Form>
+      <FormGroup>
+        <Label for="searchValue1">Prvá firma/osoba</Label>
+        <Input
+          id="searchValue1"
+          type="text"
+          value={searchValue1}
+          onChange={(e) => setSearchValue1(e.target.value)}
+          onKeyPress={checkEnter(searchConnection)}
+          placeholder="Zadaj prvú firmu / človeka"
+        />
+      </FormGroup>
+      <FormGroup>
+        <label htmlFor="searchValue2">Druhá firma/osoba</label>
+        <input
```

Comment:
There's probably a difference between `Label` and `label`, `Input` and `input`, right? You probably wanted to use strap components here :slightly_smiling_face: